### DISCUSSION TITLE
Use context get value inside of zValidator

### DISCUSSION THREAD

**Discussion by shishamo** [honojs/hono#2804] [Q&A] [answered]
Is there a way to use the context get value inside of zValidator?
I would like to use that value before/during the validation

**Comment by EdamAme-x** [comment]
hi @shishamo.

```ts
const route = app.post(
  '/posts',
  zValidator(
    'form',
    z.object({
      body: z.string(),
    }),
    (data, c) => {
        ...
    }
  ),
  (c) => {
    const validated = c.req.valid("form");
    // ... use your validated data
  }
)
```

Let's try!

**Reply by EdamAme-x** [reply]
details of function is here
https://github.com/honojs/middleware/blob/39edec1249ec3f9dae474b1abc4ebc1cf97a51c2/packages/zod-validator/src/index.ts#L12

**Reply by shishamo** [reply]
@EdamAme-x 
Thank for your help^^
I have just tried in that way but the validated value got empty
> const validated = c.req.valid("form");

**Reply by EdamAme-x** [reply]
https://hono.dev/snippets/validator-error-handling#VPSidebarNav

**Reply by EdamAme-x** [reply]
This example does not appear to return any Data.
Can you show me your code?

**Reply by EdamAme-x** [reply]
https://github.com/honojs/middleware/tree/main/packages/zod-validator

**Reply by shishamo** [reply]
@EdamAme-x 

Oh thank you i tried as the example and it is working~
Thnaks for your help!

**Comment by shishamo** [comment]
@EdamAme-x 

Oh actually not...
Maybe i found a bug or a missing feature?

The `zValidator` function looks like don't handle async function

```js
const route = app.post(
  '/posts',
  zValidator(
    'form',
    z.object({
      body: z.string(),
    }),
    async (data, c) => {
        // make some async check stuff...
    }
  ),
  (c) => {
    const validated = c.req.valid("form");
    // ... use your validated data
  }
)
```

**Reply by shishamo** [reply]
In that case 
> const validated = c.req.valid("form");

gets undefined

**Reply by EdamAme-x** [reply]
thanks @shishamo .
Could you please create the issue?
This may be a bug, I'll think about it.

**Reply by EdamAme-x** [reply]
![image](https://github.com/honojs/hono/assets/121654029/122664f6-1077-4e9a-a64d-a22ce444f2ce)

**Reply by EdamAme-x** [reply]
https://github.com/honojs/middleware/pull/552

**Reply by EdamAme-x** [reply]
@shishamo

**Reply by shishamo** [reply]
@EdamAme-x 3
Thanks for your promp action!
LGTM~

**Reply by EdamAme-x** [reply, marked as answer]
hi @shishamo 
resolved this problem by
https://github.com/honojs/middleware/pull/552

**Reply by shishamo** [reply]
thanks a lot!

**Reply by EdamAme-x** [reply]
@shishamo 
You can end with “mark as answer.”
